src/ChainSafe.Gaming.MetaMask.Unity/MetaMaskSigner.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Threading.Tasks;
+using ChainSafe.Gaming.Evm.Signers;
+using ChainSafe.Gaming.Web3.Core;
+using ChainSafe.Gaming.Web3.Core.Evm;
+using ChainSafe.Gaming.Web3.Environment;
+using Nethereum.RPC.AccountSigning;
+using Nethereum.Unity.Rpc;
+using Newtonsoft.Json;
+
+namespace ChainSafe.Gaming.MetaMask.Unity
+{
+    /// <summary>
+    /// Implementation of <see cref="ISigner"/> for Metamask.
+    /// </summary>
+    public class MetaMaskSigner : ISigner, ILifecycleParticipant
+    {
+        private readonly IMetaMaskProvider metaMaskProvider;
+        private readonly ILogWriter logWriter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetaMaskSigner"/> class.
+        /// </summary>
+        /// <param name="metaMaskProvider">Metamask provider that connects to Metamask and makes JsonRPC requests.</param>
+        /// <param name="logWriter">Log Writer used for logging messages and errors.</param>
+        public MetaMaskSigner(IMetaMaskProvider metaMaskProvider, ILogWriter logWriter)
+        {
+            this.metaMaskProvider = metaMaskProvider;
+            this.logWriter = logWriter;
+        }
+
+        /// <summary>
+        /// Signer's public key/address.
+        /// </summary>
+        public string Address { get; private set; }
+
+        /// <summary>
+        /// Implementation of <see cref="ILifecycleParticipant.WillStartAsync"/>.
+        /// Lifetime event method, called during initialization.
+        /// </summary>
+        /// <returns>async awaitable task.</returns>
+        public async ValueTask WillStartAsync()
+        {
+            Address = await metaMaskProvider.Connect();
+
+            logWriter.Log($"Connected to MetaMask account {Address}");
```

Comment:
Don't forget to remove/comment-out those logs before shipping.

Reply:
I think this is a necessary log for Devs to know, or are we logging a successful connection twice?

Reply:
Yep we're logging twice, will remove

Reply:
removed

Reply:
I mean all of the logs, not just this one. I would only keep the most important one or two like "Wallet connected", "Transaction Sent" and "Message Signed", not to spam in user's console.

Reply:
Understood, I've removed logs I believe to unnecessary/redundant.